packages/bff/src/services/purposeService.ts

```diff
@@ -4,16 +4,277 @@ import {
   PurposeVersionId,
   unsafeBrandId,
 } from "pagopa-interop-models";
-import { bffApi } from "pagopa-interop-api-clients";
-import { PagoPAInteropBeClients } from "../providers/clientProvider.js";
-import { purposeNotFound } from "../model/domain/errors.js";
-import { BffAppContext } from "../utilities/context.js";
+import {
+  agreementApi,
+  bffApi,
+  catalogApi,
+  purposeApi,
+  tenantApi,
+} from "pagopa-interop-api-clients";
+import {
+  AgreementProcessClient,
+  AuthorizationProcessClient,
+  CatalogProcessClient,
+  PurposeProcessClient,
+  TenantProcessClient,
+} from "../providers/clientProvider.js";
+import {
+  agreementNotFound,
+  eServiceDescriptorNotFound,
+  eServiceNotFound,
+  purposeDraftVersionNotFound,
+  purposeNotFound,
+  tenantNotFound,
+} from "../model/domain/errors.js";
+import { BffAppContext, Headers } from "../utilities/context.js";
+import { getLatestAgreement } from "./agreementService.js";
+import { getAllClients } from "./authorizationService.js";
+
+export const getCurrentVersion = (
+  purposeVersions: purposeApi.PurposeVersion[]
+): purposeApi.PurposeVersion | undefined => {
+  const statesToExclude: purposeApi.PurposeVersionState[] = [
+    "WAITING_FOR_APPROVAL",
+    "REJECTED",
```

Comment:
You could use values from the enum:

```suggestion
        purposeApi.PurposeVersionState.Values.WAITING_FOR_APPROVAL,
        purposeApi.PurposeVersionState.Values.REJECTED,
```